src/Servant/ConcTest/Utils.hs

```diff
@@ -0,0 +1,37 @@
+module Servant.ConcTest.Utils where
+
+-- | Generate all possible interleavings of the input lists.
+-- From https://stackoverflow.com/a/41929156
+-- NOTE: is there a more efficient way to do this?
+interleavings :: [[a]] -> [[a]]
```

Comment:
could you add an example of the input and the output?, I'm feel so stupid, but is lot more easier to seeing it that way.